dbms/src/Interpreters/Join.cpp

```diff
@@ -187,6 +180,13 @@ Join::Join(
     if (unlikely(!err.empty()))
         throw Exception("Validate join conditions error: {}" + err);
 
+    hash_join_spill_context = std::make_shared<HashJoinSpillContext>(build_spill_config, probe_spill_config, max_bytes_before_external_join, log);
+    if (hash_join_spill_context->isSpillEnabled() && restore_round >= 4)
+    {
+        LOG_INFO(log, fmt::format("restore round reach to 4, spilling will be disabled."));
```

Comment:
```suggestion
        LOG_WARN(log, fmt::format("restore round reach to 4, spilling will be disabled."));
```
maybe warn level more suitable